lib/openhab/core/things/profile_callback.rb

```diff
@@ -14,9 +14,10 @@ module ProfileCallback
         # @param [Command] command
         #
         def handle_command(command)
-          @dummy_channel_item ||= DSL::Items::ItemBuilder.item_factory.create_item(link.channel.accepted_item_type,
-                                                                                   "")
-          command = @dummy_channel_item.format_command(command)
+          link.channel.accepted_item_type&.then do |item_type|
```

Comment:
This seems a bit complicated now. And it has to traverse the link and channel on every call. Instead, can you do `unless instance_variable_defined?(:@dummy_channel_item)...`, and then move the final format call out of that conditional (you won't need an ||= anymore then).

You might as well implement the optimization of using the linked item if it's the correct time now while you're in here.